### DISCUSSION TITLE
API to open commissioning window for a server node

### DISCUSSION THREAD

**Discussion by digitaldan** [matter-js/matter.js#1420] [Feature requests (vote via thumbs up)]
This is a feature request based on a discord discussion.  When exposing devices with matter.js, for example creating a  bridged device to expose non matter devices,  there is no way to programmatically request a pairing code once the bridge has been commissioned with a 3rd party client.  To do this you must rely on that 3rd party client to open a commissioning window and provide pairing codes.

It would be very beneficial if we could do this programmatically from a server node within matter.js, so users can request additional pairing codes from within what ever application is built on top of matter.js and not have to use a 3rd party client.  The Phillips Hue matter bridge for example provides similar functionality from within their app.

**Comment by Apollon77** [comment]
In fact the currently working solution is

```javascript
const comm = node.env.get(DeviceCommissioner);
await comm.allowBasicCommissioning(() => {
    // End callback
});
```
